core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -549,11 +549,9 @@ public Map<SchemaTableName, RelationType> getRelationTypes(Session session, Qual
 
         Optional<QualifiedObjectName> objectName = prefix.asQualifiedObjectName();
         if (objectName.isPresent()) {
-            Optional<RelationType> relationType = getRelationTypeIfExists(session, objectName.get());
-            if (relationType.isPresent()) {
-                return ImmutableMap.of(objectName.get().asSchemaTableName(), relationType.get());
-            }
-            return ImmutableMap.of();
+            return getRelationTypeIfExists(session, objectName.get())
+                    .map(type -> ImmutableMap.of(objectName.get().asSchemaTableName(), type))
+                    .orElseGet(ImmutableMap::of);
```

Comment:
`ImmutableMap.of()` is a singleton (ie: has no allocation overhead), so `.orElse(ImmutableMap.of())` is cheaper than `.orElseGet(ImmutableMap::of)`